### PR TITLE
Kr/curator dashboard apis

### PR DESCRIPTION
**Pull request by KanishkaRajputd**
## Summary by CodeRabbit

- **New Features**
	- Introduced multiple API endpoints for editing curator bios, retrieving curated bounties, and fetching curator information.
	- Enhanced querying capabilities for child bounties with additional parameters.

- **Bug Fixes**
	- Improved handling of claimed amounts for child bounties.

- **Documentation**
	- Added new success messages for curator bio edits.

- **Refactor**
	- Updated GraphQL queries for better filtering based on curator and status.
	- Enhanced structure of child bounty interface with an optional property.

### CODE REVIEW

File: pages/api/v1/bounty/curator/getAllCuratedBountiesAndChildBounties.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	page: number;
+	network: string;
+	curatorAddress: string;
+}
+
+export async function getAllBounties({ page, network, curatorAddress }: Args): Promise<IApiResponse<{ bounties: IBountyListing[]; totalBountiesCount: number }>> {
+	try {
+		if (!network || !isValidNetwork(network)) throw apiErrorWithStatusCode(messages.INVALID_NETWORK, 400);
+
+		if (isNaN(page) || !curatorAddress?.length) throw apiErrorWithStatusCode(messages.INVALID_PARAMS, 400);
```
Comment (alphainfinitus) [inline pages/api/v1/bounty/curator/getAllCuratedBountiesAndChildBounties.ts:45]: check missing for valid curator address

Comment (alphainfinitus) [reply pages/api/v1/bounty/curator/getAllCuratedBountiesAndChildBounties.ts:45]: please check with getSubstrateAddress

File: pages/api/v1/bounty/curator/getAllCuratedBountiesAndChildBounties.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+		let totalBounties = 0;
+
+		const encodedCuratorAddress = getEncodedAddress(curatorAddress, network);
+
+		const subsquidBountiesRes = await fetchSubsquid({
+			network,
+			query: GET_ALL_BOUNTIES,
+			variables: {
+				curator_eq: encodedCuratorAddress,
+				limit: 10,
```
Comment (alphainfinitus) [inline pages/api/v1/bounty/curator/getAllCuratedBountiesAndChildBounties.ts:56]: hard coded listing limit

File: pages/api/v1/bounty/curator/getCuratorGeneralInfo.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+	return rewardsByMonth;
+}
+
+const handler: NextApiHandler<IRes | MessageType> = async (req, res) => {
+	storeApiKeyUsage(req);
+
+	const network = String(req.headers['x-network']);
+
+	const { userAddress } = req.body;
```
Comment (alphainfinitus) [inline pages/api/v1/bounty/curator/getCuratorGeneralInfo.ts:76]: please verify if valid address

File: pages/api/v1/bounty/curator/getCuratorGeneralInfo.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+				variables: {
+					curator_eq: encodedCuratorAddress,
+					parentBountyIndex_eq: subsquidBounty?.index
+				}
+			});
+
+			const subsquidChildBountyData = subsquidChildBountiesRes?.data?.proposals || [];
+
+			let claimedAmount = ZERO_BN;
+			const totalChildBountiesCount = 0;
```
Comment (alphainfinitus) [inline pages/api/v1/bounty/curator/getCuratorGeneralInfo.ts:127]: this constant doesn't update

File: src/queries.ts [hunk 5]
```diff
@@ -821,17 +821,18 @@ query ChildBountiesByParentIndex($parentBountyIndex_eq: Int = 11, $limit: Int, $
 }
 `;
 
-export const GET_ALL_CHILD_BOUNTIES_BY_PARENT_INDEX = `query ChildBountiesByParentIndex($parentBountyIndex_eq: Int = 11) {
-  proposalsConnection(orderBy: createdAtBlock_DESC, where: {parentBountyIndex_eq: $parentBountyIndex_eq, type_eq: ChildBounty}) {
+export const GET_ALL_CHILD_BOUNTIES_BY_PARENT_INDEX = `query ChildBountiesByParentIndex($parentBountyIndex_eq: Int, $curator_eq: String, $status_eq: ProposalStatus ) {
```
Comment (alphainfinitus) [inline src/queries.ts:824]: please make sure changes to this query does not effect other code where this query is being imported

**Review by alphainfinitus — CHANGES_REQUESTED** [review]

**Review by AleemAlam — CHANGES_REQUESTED** [review]

**Pull request merged**